libraries/networking/src/udt/SequenceNumber.h

```diff
@@ -0,0 +1,149 @@
+//
+//  SequenceNumber.h
+//  libraries/networking/src/udt
+//
+//  Created by Clement on 7/23/15.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#ifndef hifi_SequenceNumber_h
+#define hifi_SequenceNumber_h
+
+#include <functional>
+
+#include <glm/detail/func_common.hpp>
+
+namespace udt {
+
+class SequenceNumber {
+public:
+    // Base type of sequence numbers
+    using Type = int32_t;
+    using UType = uint32_t;
+    
+    // Values are for 29 bit SequenceNumber
+    static const Type THRESHOLD = 0x0FFFFFFF; // threshold for comparing sequence numbers
+    static const Type MAX = 0x1FFFFFFF; // maximum sequence number used in UDT
+    
+    SequenceNumber() = default;
+    SequenceNumber(const SequenceNumber& other) : _value(other._value) {}
+    
+    // Only explicit conversions
+    explicit SequenceNumber(char* value) { _value = (*reinterpret_cast<int32_t*>(value)) & MAX; }
+    explicit SequenceNumber(Type value) { _value = (value <= MAX) ? ((value >= 0) ? value : 0) : MAX; }
+    explicit SequenceNumber(UType value) { _value = (value <= MAX) ? value : MAX; }
+    explicit operator Type() { return _value; }
+    explicit operator UType() { return static_cast<UType>(_value); }
+    
+    inline SequenceNumber& operator++() {
+        _value = (_value == MAX) ? 0 : ++_value;
```

Comment:
We can avoid the branch here and other cases by using `_value = (_value + 1) % (MAX + 1)` or `_value = (_value + 1) & MAX`